contracts/interactiveAdapters/Amun/AmunLiquidityInteractiveAdapter.sol

```diff
@@ -0,0 +1,98 @@
+// Copyright (C) 2020 Zerion Inc. <https://zerion.io>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+//
+// SPDX-License-Identifier: LGPL-3.0-only
+
+pragma solidity 0.7.3;
+pragma experimental ABIEncoderV2;
+
+import { ERC20 } from "../../shared/ERC20.sol";
+import { SafeERC20 } from "../../shared/SafeERC20.sol";
+import { TokenAmount } from "../../shared/Structs.sol";
+import { ERC20ProtocolAdapter } from "../../adapters/ERC20ProtocolAdapter.sol";
+import { InteractiveAdapter } from "../InteractiveAdapter.sol";
+import { IVault } from "../../interfaces/IVault.sol";
+
+/**
+ * @title Interactive adapter for AmunLiquidity.
+ * @dev Implementation of InteractiveAdapter abstract contract.
+ * @author Timo <Timo@amun.com>
+ */
+contract AmunLiquidityInteractiveAdapter is InteractiveAdapter, ERC20ProtocolAdapter {
+    using SafeERC20 for ERC20;
+
+    /**
+     * @notice Deposits tokens to the Yearn Vault.
+     * @param tokenAmounts Array with one element - TokenAmount struct with
+     * underlying token address, underlying token amount to be deposited, and amount type.
+     * @param data ABI-encoded additional parameters:
+     *     - liquidityToken - liquidityToken address.
+     * @return tokensToBeWithdrawn Array with ane element - liquidityToken.
+     * @dev Implementation of InteractiveAdapter function.
+     */
+    function deposit(TokenAmount[] calldata tokenAmounts, bytes calldata data)
+        external
+        payable
+        override
+        returns (address[] memory tokensToBeWithdrawn)
+    {
+        require(tokenAmounts.length == 1, "LBIA: should be 1 tokenAmount[1]");
+
+        address liquidityToken = abi.decode(data, (address));
+
+        address token = tokenAmounts[0].token;
+        uint256 amount = getAbsoluteAmountDeposit(tokenAmounts[0]);
+
+        tokensToBeWithdrawn = new address[](1);
+        tokensToBeWithdrawn[0] = liquidityToken;
+
+        ERC20(token).safeApprove(liquidityToken, amount, "LBIA");
+        // solhint-disable-next-line no-empty-blocks
+        try IVault(liquidityToken).deposit(amount, 101)  {} catch Error(string memory reason) {
```

Comment:
101 as a magical number.

Recommend adding it to a variable name above:

`uint8 referralCode = 101`